dcnum/feat/feat_moments/mt_legacy.py

```diff
@@ -11,67 +11,99 @@ def moments_based_features(mask, pixel_size):
     size = mask.shape[0]
 
     empty = np.full(size, np.nan, dtype=np.float64)
-    deform = np.copy(empty)
+
+    # features from raw contour
     size_x = np.copy(empty)
     size_y = np.copy(empty)
-    pos_x = np.copy(empty)
-    pos_y = np.copy(empty)
+    aspect = np.copy(empty)
     area_msd = np.copy(empty)
+    area_um_raw = np.copy(empty)
+    per_um_raw = np.copy(empty)
+    deform_raw = np.copy(empty)
     area_ratio = np.copy(empty)
-    area_um = np.copy(empty)
-    aspect = np.copy(empty)
-    tilt = np.copy(empty)
-    inert_ratio_cvx = np.copy(empty)
+    per_ratio = np.copy(empty)
     inert_ratio_raw = np.copy(empty)
     inert_ratio_prnc = np.copy(empty)
+    eccentr_prnc = np.copy(empty)
+    tilt = np.copy(empty)
+    s_x = np.copy(empty)
+    s_y = np.copy(empty)
+
+    # features from convex hull
+    pos_x = np.copy(empty)
+    pos_y = np.copy(empty)
+    area_um = np.copy(empty)
+    deform = np.copy(empty)
+    inert_ratio_cvx = np.copy(empty)
     # The following valid-array is not a real feature, but only
     # used to figure out which events need to be removed due
     # to invalid computed features, often due to invalid contours.
     valid = np.full(size, False)
 
     for ii in range(size):
-        cont_raw = contour_single_opencv(mask[ii])
+        # raw contour
+        cont_raw = contour_single_opencv(mask[ii]).astype("float32")
```

Comment:
Please explain why you are casting to float32.

Reply:
it is something that is needed for correct porosity and def calculation once the contour is not give by integer number as in case of smoothed contour.  Here it is not necessary, indeed. I can change it back.

Reply:
It is fine as long as you add this info as a comment. Why float32 and not float64?